File: Data Structures & Algorithms/max-water-container/submission-0.java
class Solution {
    public int maxArea(int[] heights) {
        int max = 0;
        int i = 0;
        int j = heights.length - 1;
        while (i < j) {
            int vol = Math.min(heights[i], heights[j]) * (j - i);
            max = Math.max(vol, max);
            System.out.println(max);
            if (heights[i] > heights[j]) {
                j--;
            }
            else {
                i++;
            }
        }
        return max;
    }
}

File: Data Structures & Algorithms/insert-new-interval/submission-0.py
class Solution:
    def insert(self, intervals: List[List[int]], newInterval: List[int]) -> List[List[int]]:
        res = []
        curr = newInterval
        for i in intervals:
            if curr[0] > i[1]:
                res.append(i)
            elif curr[1] < i[0]:
                res.append(curr)
                curr = i
            else:
                curr = [min(curr[0], i[0]), max(curr[1], i[1])]
        
        res.append(curr)

        return res